components/speedreader/speedreader_rewriter_service.cc

```diff
@@ -36,7 +40,13 @@ std::string GetDistilledPageStylesheet(const base::FilePath& stylesheet_path) {
         IDR_SPEEDREADER_STYLE_DESKTOP);
   }
 
-  return "<style id=\"brave_speedreader_style\">" + stylesheet + "</style>";
+  const std::string style_hash = crypto::SHA256HashString(stylesheet);
+  const std::string style_hash_b64 = base::Base64Encode(base::as_bytes(
+    base::make_span(style_hash)));
+
+  return "<meta http-equiv=\"Content-Security-Policy\" content=\""
+            "script-src 'none'; style-src 'sha256-" + style_hash_b64 + "'\">\n"
+         "<style id=\"brave_speedreader_style\">" + stylesheet + "</style>";
```

Comment:
This won't apply to the initial load of the built-in stylesheet in the `SpeedreaderRewriterService` constructor. Probably this code should move to a `WrapStylesheet` helper which is called both here and when `content_stylesheet_` is set directly in the constructor.

Reply:
okay i can change accordingly, though i've yet to find a case where the `content_stylesheet_` property is used ;)

Reply:
done and done https://github.com/brave/brave-core/pull/11545/commits/3c5fb368e1e3606015bee801e019a9727367aa1d